Cannot build old scipy with new numpy

`RUN pip install --no-binary scipy -U "%s"` fails with the config
```
--- Shuffle Parameters ---
numpy: 1.16
scipy: 0.19
base: ubuntu14_py36-pyenv
cuda_cudnn_nccl: ('cuda90', 'cudnn7-cuda9', 'nccl2.3-cuda9')
--- Configuration ---
nccl: nccl2.3-cuda9
cudnn: cudnn7-cuda9
base: ubuntu14_py36-pyenv
requires: ['numpy<1.17', 'scipy<0.20']
cuda: cuda90
```

A tail of the log:
```
    building extension "scipy.linalg._flapack" sources
    from_template:> build/src.linux-x86_64-3.6/scipy/linalg/flapack.pyf
    Including file scipy/linalg/flapack_user.pyf.src
    Mismatch in number of replacements (base <prefix=s,d,c,z>) for <__l1=c,z>. Ignoring.
    Mismatch in number of replacements (base <prefix=s,d,c,z>) for <__l1=c,z>. Ignoring.
    Mismatch in number of replacements (base <prefix2=s,d>) for <__l1=S/D>. Ignoring.
    Mismatch in number of replacements (base <prefix2c=c,z>) for <__l1=C/Z>. Ignoring.
    Mismatch in number of replacements (base <prefix2=s,d>) for <__l1=S/D>. Ignoring.
    Mismatch in number of replacements (base <prefix2c=c,z>) for <__l1=C/Z>. Ignoring.
    Mismatch in number of replacements (base <prefix2=s,d>) for <__l1=S/D>. Ignoring.
    Mismatch in number of replacements (base <prefix2=s,d>) for <__l1=S/D>. Ignoring.
    Mismatch in number of replacements (base <prefix2c=c,z>) for <__l1=C/Z>. Ignoring.
    Mismatch in number of replacements (base <prefix2c=c,z>) for <__l1=C/Z>. Ignoring.
    f2py options: []
    f2py: build/src.linux-x86_64-3.6/scipy/linalg/flapack.pyf
    Reading fortran codes...
    	Reading file 'build/src.linux-x86_64-3.6/scipy/linalg/flapack.pyf' (format:free)
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-gvxg_fya/scipy/setup.py", line 416, in <module>
        setup_package()
      File "/tmp/pip-install-gvxg_fya/scipy/setup.py", line 412, in setup_package
        setup(**metadata)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/distutils/core.py", line 171, in setup
        return old_setup(**new_attr)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/setuptools/__init__.py", line 145, in setup
        return distutils.core.setup(**attrs)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/distutils/dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/distutils/command/install.py", line 62, in run
        r = self.setuptools_run()
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/distutils/command/install.py", line 36, in setuptools_run
        return distutils_install.run(self)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/distutils/command/install.py", line 545, in run
        self.run_command('build')
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/distutils/command/build.py", line 47, in run
        old_build.run(self)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/distutils/command/build.py", line 135, in run
        self.run_command(cmd_name)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/distutils/command/build_src.py", line 148, in run
        self.build_sources()
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/distutils/command/build_src.py", line 165, in build_sources
        self.build_extension_sources(ext)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/distutils/command/build_src.py", line 325, in build_extension_sources
        sources = self.f2py_sources(sources, ext)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/distutils/command/build_src.py", line 542, in f2py_sources
        + ['--build-dir', target_dir, source])
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/f2py/f2py2e.py", line 425, in run_main
        postlist = callcrackfortran(files, options)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/f2py/f2py2e.py", line 329, in callcrackfortran
        postlist = crackfortran.crackfortran(files)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/f2py/crackfortran.py", line 3251, in crackfortran
        readfortrancode(files, crackline)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/f2py/crackfortran.py", line 501, in readfortrancode
        dowithline(finalline)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/f2py/crackfortran.py", line 796, in crackline
        analyzeline(m, pat[1], line)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/f2py/crackfortran.py", line 1098, in analyzeline
        last_name = updatevars(typespec, selector, attr, edecl)
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/f2py/crackfortran.py", line 1542, in updatevars
        attrspec = [x.strip() for x in markoutercomma(attrspec).split('@,@')]
      File "/opt/pyenv/versions/3.6.5/lib/python3.6/site-packages/numpy/f2py/crackfortran.py", line 833, in markoutercomma
        assert not f, repr((f, line, l))
    AssertionError: (-1, 'intent(in))', 'intent(in))')
    
    ----------------------------------------
Command "/opt/pyenv/versions/3.6.5/bin/python3.6 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-gvxg_fya/scipy/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-m5e3kckm/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-install-gvxg_fya/scipy/
```

--- a/shuffle.py
+++ b/shuffle.py
@@ -77,6 +77,11 @@ def get_shuffle_params(params, index):
             if ret['numpy'] in ['1.9', '1.10', '1.11', '1.12']:
                 ret['numpy'] = '1.13'
 
+    # SciPy 0.19 installation from source (--no-binary) fails with new NumPy
+    if ret['numpy'] in ['1.16']:
+        if ret.get('scipy', None) in ['0.18', '0.19']:
+            ret['scipy'] = '1.0'
+
     # Theano 1.0.3 or earlier does not support NumPy 1.16
     if ret.get('theano') in ['0.8', '0.9']:
         if ret['numpy'] == '1.16':

